recipes/pybind11/all/test_package/conanfile.py

```diff
@@ -1,14 +1,19 @@
 from conans import ConanFile, CMake, tools
 import os
 import sys
-from platform import python_version
 
 
 class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
     generators = "cmake", "cmake_find_package"
 
     def build(self):
+        tools.replace_in_file(os.path.join(self.build_folder, "Findpybind11.cmake"),
+                              "set_property(TARGET pybind11::pybind11 APPEND PROPERTY",
+                              "")
+        tools.replace_in_file(os.path.join(self.build_folder, "Findpybind11.cmake"),
+                              "INTERFACE_LINK_LIBRARIES \"${pybind11_COMPONENTS}\")",
+                              "")
```

Comment:
This change is only here to make the POC work. We would consider this as a feature for the Conan client in the ``cmake`` generator to make the "main" component (``pybind11::pybind11`` in this case) totally overrideable.